src/conn/conn_layered.c

```diff
@@ -0,0 +1,1861 @@
+ */
+int
+__wt_disagg_copy_metadata_process(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_DISAGG_COPY_METADATA *entry, *tmp;
+
+    conn = S2C(session);
+
+    __wt_spin_lock(session, &conn->disaggregated_storage.copy_metadata_lock);
+
+    WT_TAILQ_SAFE_REMOVE_BEGIN(entry, &conn->disaggregated_storage.copy_metadata_qh, q, tmp)
+    {
+        WT_ERR(__disagg_copy_shared_metadata_one(session, entry->stable_uri));
+        WT_ERR(__wt_disagg_copy_shared_metadata_layered(session, entry->table_name));
+
+        TAILQ_REMOVE(&conn->disaggregated_storage.copy_metadata_qh, entry, q);
+        __wt_free(session, entry->stable_uri);
+        __wt_free(session, entry->table_name);
+        __wt_free(session, entry);
+    }
+    WT_TAILQ_SAFE_REMOVE_END
+
+err:
+    __wt_spin_unlock(session, &conn->disaggregated_storage.copy_metadata_lock);
+
+    return (ret);
+}
+
+/*
+ * __disagg_metadata_table_init --
+ *     Initialize the shared metadata table.
+ */
+static int
+__disagg_metadata_table_init(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *internal_session;
+
+    conn = S2C(session);
+
+    WT_ERR(__wt_open_internal_session(conn, "disagg-init", false, 0, 0, &internal_session));
+    WT_ERR(__wt_session_create(
+      internal_session, WT_DISAGG_METADATA_URI, "key_format=S,value_format=S,log=(enabled=false)"));
+
+err:
+    if (internal_session != NULL)
+        WT_TRET(__wt_session_close_internal(internal_session));
+    return (ret);
+}
+
+/*
+ * __disagg_set_last_materialized_lsn --
+ *     Set the latest materialized LSN.
+ */
+static int
+__disagg_set_last_materialized_lsn(WT_SESSION_IMPL *session, uint64_t lsn)
+{
+    WT_DISAGGREGATED_STORAGE *disagg;
+    uint64_t cur_lsn;
+
+    disagg = &S2C(session)->disaggregated_storage;
+    WT_ACQUIRE_READ(cur_lsn, disagg->last_materialized_lsn);
+
+    if (cur_lsn > lsn)
+        return (EINVAL); /* Can't go backwards. */
+
+    WT_RELEASE_WRITE(disagg->last_materialized_lsn, lsn);
+    return (0);
+}
+
+/*
+ * __wti_disagg_conn_config --
+ *     Parse and setup the disaggregated server options for the connection.
+ */
+int
+__wti_disagg_conn_config(WT_SESSION_IMPL *session, const char **cfg, bool reconfig)
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_NAMED_PAGE_LOG *npage_log;
+    uint64_t checkpoint_id, complete_checkpoint, next_checkpoint_id, open_checkpoint;
+    bool leader, was_leader;
+
+    conn = S2C(session);
+    leader = was_leader = conn->layered_table_manager.leader;
+    npage_log = NULL;
+
+    /* Reconfigure-only settings. */
+    if (reconfig) {
+
+        /* Pick up a new checkpoint (followers only). */
+        WT_ERR(__wt_config_gets(session, cfg, "disaggregated.checkpoint_meta", &cval));
+        if (cval.len > 0) {
+            /*
+             * TODO: currently the leader silently ignores the checkpoint_meta configuration as it
```

Comment:
@quchenhao does need to still be a `TODO`? If so, could you please create a ticket?

Reply:
Yes, I think this is still a todo.

Reply:
Thanks!